python/ray/serve/_private/http_proxy.py

```diff
@@ -703,6 +707,21 @@ async def send_request_to_replica_streaming(
         return status_code
 
 
+class RequestIdMiddleware:
+    def __init__(self, app):
+        self.app = app
+
+    async def __call__(self, scope, receive, send):
+        async def send_with_request_id(message):
+            if message["type"] == "http.response.start":
+                request_id = ray.serve.context._serve_request_context.get().request_id
+                headers = MutableHeaders(scope=message)
```

Comment:
does this update the underlying `scope` passed in?

Reply:
Not sure i understand the question, I assume all headers information for `http.response.start` is under the message directly. 
The `MutableHeaders` will only extract headers from the message https://github.com/encode/starlette/blob/2168e47052239da5df35d5353bb986f760c51cef/starlette/datastructures.py#L534

Reply:
oh I'm just confused because you're never modifying `scope` in the message, does `headers.append(RAY_SERVE_REQUEST_ID, request_id)` do it?

Reply:
ah, yes. the `headers` object hold the message header Dict ref. and whenever you update the `headers`, it will update the message. It is same as you update the message header directly, but this is more standard instead of updating raw dict by yourself.

Reply:
perfect thanks for explaining